app/services/collect_inventories_service.rb

```diff
@@ -0,0 +1,44 @@
+class CollectInventoriesService
+  attr_reader :inventory_ids
+
+  def initialize(options)
+    @options = options.deep_symbolize_keys
+    @service_offering_id = @options[:service_offering_id]
+    @task_id = @options[:task_id]
+    @inventory_ids = []
+  end
+
+  def process
+    Task.update(@task_id, :state => "running", :status => "ok")
+
+    visited = []
+    collect_inventory(@service_offering_id, visited, @inventory_ids)
+
+    Task.update(@task_id, :state => "completed", :status => "ok", :output => {:applied_inventories => @inventory_ids})
```

Comment:
@hsong-rh We dont need a Task object for this, the REST API can just return an Array of Tags